Added support for Cache VOL

zhenghh04 opened the pull request:
This allows h5bench to pause the async operations in Cache VOL and start them later after issuing out all the dataset write calls.

jeanbez commented:
@zhenghh04 could you also add the documentation for this?

sbyna commented:
@zhenghh04 Could you please add / update documentation on how to use the benchmark with Cache VOL?

zhenghh04 commented:
Main changes: 
- h5bench_write.c: please check regions with -DUSE_CACHE_VOL
- h5bench_util.c: please check regions with -DUSE_CACHE_VOL
- h5bench:
    - added json parser for Cache VOL.
    - enabling vol as long as vol[‘connector'] is not empty (previously it requires async vol to be set)
- CMakeLists.txt: included Cache build instruction support
- docs: added related documentation for cache VOL (build instruction, setup, etc)
- .github/workflows/h5bench-hdf5-develop.yml: added Cache VOL building, h5bench + cache vol building, testing

jeanbez commented:
@sbyna  and @houjun if you have some time available, could you help us double-check this new feature?
@zhenghh04 I've included some additional test cases and checked the changes, it seems good to me.

jeanbez commented on h5bench_patterns/h5bench_write.c:
```diff
@@ -756,7 +757,10 @@ _run_benchmark_write(bench_params params, hid_t file_id, hid_t fapl, hid_t files
             printf("Writing %s ... \n", grp_name);
-
+#ifdef USE_CACHE_VOL
+        MPI_Barrier(MPI_COMM_WORLD);
```
@zhenghh04 can you confirm this?

zhenghh04 commented:
@jeanbez @houjun I pushed a new commit: 

I added back "MODE" option in cfg file for Cache VOL support. If the user does not set MODE in the cfg file, then it will be as it used to be: i.e., determined merely by whether async vol connector is used or not. 

I also removed the barrier.
